### PR TITLE
[FLOC-2736] REST API endpoint: list leases

### PR DESCRIPTION
**Pull request by itamarst**
1. Expand schema to know about leases. The tests were structured so the good/bad examples can be re-used in later schema tests.
2. Add a new endpoint for listing leases.

### CODE REVIEW

**Comment by itamarst** [comment]
Missing docs, will add those tomorrow I guess.

File: flocker/control/test/test_httpapi.py [hunk 1]
```diff
@@ -42,6 +45,7 @@
 from .._clusterstate import ClusterStateService
 from .._config import (
     FlockerConfiguration, FigConfiguration, model_from_configuration)
+from .._model import Leases
```
Comment (dwgebler) [inline flocker/control/test/test_httpapi.py]: Minor quibble, but the rest of our models used in these tests are imported from module level defined in `__init__.py`

File: flocker/control/schema/types.yml [hunk 2]
```diff
@@ -280,3 +280,36 @@ definitions:
...
+
+  lease_expiration:
+    title: "Lease Expiration"
+    description: |
+      The number of seconds until a lease expires, or null to indicate no expiration.
+    type:
+      - "number"
+      - "null"
+
+  lease_dataset_id:
```
Comment (dwgebler) [inline flocker/control/schema/types.yml]: Any reason to not reuse the existing `dataset_id` type definition?

**Comment by dwgebler** [comment]
Couple of minor inline comments to address or answer, otherwise please resubmit with the docs when they've been added. Thanks.

File: flocker/control/schema/types.yml (hunk 2 above)
Comment (itamarst) [inline flocker/control/schema/types.yml]: The prose doesn't quite make sense in this API endpoint's context, alas.

Comment (itamarst) [inline flocker/control/schema/types.yml]: Which suggests a bigger problem but not one I'm going to think too hard about for now.

**Comment by dwgebler** [comment]
Thanks, LGTM when the build passes.

**Pull request merged**